chef/cookbooks/nova_dashboard/recipes/server.rb

```diff
@@ -17,6 +17,31 @@
 include_recipe "apache2::mod_wsgi"
 include_recipe "apache2::mod_rewrite"
 
+keystone_settings = KeystoneHelper.keystone_settings(node, @cookbook_name)
+
+# Multidomain can only be enabled if the API version of Keystone is
+# equal or bigger that v3.
+keystone_api_version = keystone_settings['api_version']
+keystone_internal_auth_url = keystone_settings['internal_auth_url']
+
+# TODO(aplanas): Remove this line when keystone API 'v3' works for
+# every OpenStack configuration file.
+if node["nova_dashboard"]["multidomain"] && keystone_api_version.to_f < 3
+  keystone_api_version = '3'
+  keystone_node = KeystoneHelper.search_for_keystone(node, @cookbook_name)
+  keystone_internal_auth_url = KeystoneHelper.versioned_service_URL(
+    keystone_node, keystone_node[:fqdn],
```

Comment:
I think you should be using `internal_url_host` in keystone_settings, not the fqdn.

In general, we've really tried to make sure that you don't need the keystone node and can use `keystone_settings` instead, and I'm not sure we should go back to requiring the node. Maybe we should instead change the `service_url` bit to not require passing a node?